lib/ansible/plugins/lookup/cyberarkpassword.py

```diff
@@ -0,0 +1,142 @@
+# (c) 2017, Edward Nunez <edward.nunez@cyberark.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+import subprocess
+
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+
+CLIPASSWORDSDK_CMD = '/opt/CARKaim/sdk/clipasswordsdk'
+if os.getenv('AIM_CLIPASSWORDSDK_CMD') is not None:
+    CLIPASSWORDSDK_CMD = os.environ['AIM_CLIPASSWORDSDK_CMD']
+
+
+class CyberarkPassword:
+
+    def __init__(self, **kwargs):
+
+        self.appid = kwargs.pop('appid', None)
+        self.query = kwargs.pop('query', None)
+        self.output = kwargs.pop('output', None)
+
+        # Support for Generic parameters to be able to specify
+        # FailRequestOnPasswordChange, Queryformat, Reason, etc.
+        self.extraParms = []
+        for key, value in kwargs.items():
+            self.extraParms.append('-p')
+            self.extraParms.append(key + "=" + value)
+
+        if self.appid is None:
+            raise AnsibleError("No Application ID specified")
+        if self.query is None:
+            raise AnsibleError("No Vault query specified")
+
+        if self.output is None:
+            # If no output is specified, return at least the password
+            self.output = "password"
+        else:
+            # To avoid reference issues/confusion to values, all
+            # output 'keys' will be in lowercase.
+            self.output = self.output.lower()
+
+        self.delimiter = "@#@"  # Known delimiter to split output results
+
+    def get(self):
+
+        resultDict = {}
+
+        try:
+            allParms = [
+                CLIPASSWORDSDK_CMD,
+                'GetPassword',
+                '-p',
+                'AppDescs.AppID=' +
+                self.appid,
+                '-p',
+                'Query=' +
+                self.query,
+                '-o',
+                self.output,
+                '-d',
+                self.delimiter]
+            allParms.extend(self.extraParms)
+
+            credential = subprocess.check_output(allParms)
+
+            if len(credential) > 0 and credential.endswith("\n"):
```

Comment:
```
$ python3.5
Python 3.5.3 (default, Jan 19 2017, 14:11:04) 
[GCC 6.3.0 20170118] on linux
>>> import subprocess
>>> out = subprocess.check_output('ls')
>>> out.endswith('\n')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: endswith first arg must be bytes or a tuple of bytes, not str
```
I am not sure how it could work ?